vault-util/cmd/print-git-credentials.go

```diff
@@ -2,34 +2,82 @@ package cmd
 
 import (
 	"fmt"
+	"io/ioutil"
+	"strings"
+	"time"
 
 	"github.com/chef/ci-studio-common/lib"
 	"github.com/chef/ci-studio-common/vault-util/internal/account"
 
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
+	"gopkg.in/djherbis/times.v1"
 )
 
 var (
 	gitCredentialCmd = &cobra.Command{
 		Use:   "print-git-credentials [APP]",
 		Short: "Utility that will print credentials for a GitHub App from Vault in git-credential-helper format.",
-		Run:   printCredentials,
+		RunE:  printCredentialsE,
 	}
 )
 
 func init() {
 	rootCmd.AddCommand(gitCredentialCmd)
 }
 
-func printCredentials(cmd *cobra.Command, args []string) {
+func printCredentialsE(cmd *cobra.Command, args []string) error {
 	appName := viper.GetString("github.default_app")
-	if len(args) >= 1 {
+	if len(args) == 2 {
 		appName = args[0]
 	}
 
+	filename := lib.SettingsPath(fmt.Sprintf("github_token_%s", appName))
+
+	var token string
+
+	times, err := times.Stat(filename)
```

Comment:
❓ Do we know if we can get away with just the `mtime` here? The mtime is available via the standard library without introducing a dependency with something like?

```golang
	info, err := os.Stat(filename)
	if err != nil {
		...
	} else {
		if info.ModTime().Before(someMinutesAgo) {
		...
	}
	...
```

Reply:
I reason I went with the external library is because the library handled the differences between windows and linux.

Reply:
Unless I misread the documentation and there is no difference in golang libraries for accessing mtime.